src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,648 @@
+        result = self._getFromCfgSvc(firstOption, secondOption, defaultOption)
+        if result["OK"]:
+            xrslExtraString = result["Value"]
+            self.log.debug("Found %s : %s" % (xtraVariable, xrslExtraString))
+        if xrslExtraString:
+            self.log.always("%s : %s" % (xtraVariable, xrslExtraString))
+            self.log.always(" --- to be added to pilots going to CE : %s" % self.ceHost)
+        return xrslExtraString
+
+    #############################################################################
+    def __writeXRSL(self, executableFile):
+        """Create the JDL for submission"""
+        diracStamp = makeGuid()[:8]
+        # Evaluate the number of processors to allocate
+        nProcessors = self.ceParameters.get("NumberOfProcessors", 1)
+
+        xrslMPAdditions = ""
+        if nProcessors and nProcessors > 1:
+            xrslMPAdditions = """
+(count = %(processors)u)
+(countpernode = %(processorsPerNode)u)
+%(xrslMPExtraString)s
+      """ % {
+                "processors": nProcessors,
+                "processorsPerNode": nProcessors,  # This basically says that we want all processors on the same node
+                "xrslMPExtraString": self.xrslMPExtraString,
+            }
+
+        xrsl = """
+&(executable="%(executable)s")
+(inputFiles=(%(executable)s "%(executableFile)s"))
+(stdout="%(diracStamp)s.out")
+(stderr="%(diracStamp)s.err")
+(outputFiles=("%(diracStamp)s.out" "") ("%(diracStamp)s.err" ""))
+(queue=%(queue)s)
+%(xrslMPAdditions)s
+%(xrslExtraString)s
+    """ % {
+            "executableFile": executableFile,
+            "executable": os.path.basename(executableFile),
+            "diracStamp": diracStamp,
+            "queue": self.arcQueue,
+            "xrslMPAdditions": xrslMPAdditions,
+            "xrslExtraString": self.xrslExtraString,
+        }
+
+        return xrsl, diracStamp
+
+    #############################################################################
+    def _getFromCfgSvc(self, *options):
+        """Try the possibilities in the given order - firstOption, is always the most preferred value
+        and the last one is the default Option"""
+        if len(options) < 1:
+            return S_ERROR("No paths for CS found")
+        self.log.debug("Trying to get information from the CS")
+        for option in options:
+            if len(option) < 2:
+                continue
+            self.log.debug("Trying to retrieve from CS location : %s" % option)
+            result = gConfig.getValue(option)
+            if result:
+                return S_OK(result)
+        # Likely none of the paths were valid in the CS
+        return S_OK()
+
+    def _pilot_toAPI(self, pilot):
+        # Add CE and protocol information to pilot ID
+        if "://" in pilot:
+            self.log.warning("Pilot already in API format", "%s" % pilot)
+            return pilot
+        pilotAPI = "gsiftp://" + self.ceHost + "/" + pilot
```

Comment:
It is a string representing the pilot which is stored in the pilotDB and communicated with the CE. We are not using the urllib in this module. What benefit do we get from using urllib?

Reply:
It would make sure that the string is correctly constructed (e.g. "correct" number of "/").

Reply:
Considering what we are working with, what exactly are we worried about? I mean, I can do it, but it seems unnecessary to use an external library to do something that we know exactly what to do and is simple.

Reply:
BTW, gsiftp used here?

Reply:
Apologies for missing this question until now.

No - gsiftp is not used here. It is used in a dummy fashion, only to get the pilot into the "old-style DIRAC" format. The protocol is chosen as the same protocol previously used in the ARC API, primarily so that a shifter looking at the pilot monitoring will not be too confused.

Resolving this conversation as the request is implemented in the code.